rs/utils/src/str.rs

```diff
@@ -93,7 +94,9 @@ impl StrEllipsize for str {
         }
 
         const ELLIPSIS: &str = "...";
-        assert!(max_len >= ELLIPSIS.len());
+        if max_len < ELLIPSIS.len() {
```

Comment:
NIT: maybe to continue printing dots?
```rust
        if max_len < ELLIPSIS.len() {
            return ".".repeat(max_len);
        }
```

Reply:
Great idea. Done. 

I also went for `<=` as opposed to `<` here as the case where `max_len == ELLIPSIS.len()` is also just printing dots.

Reply:
Reverted to empty string as per [the discussion below](https://github.com/dfinity/ic/pull/4549#discussion_r2016282804).